### PR TITLE
Add help to only check items that you have completed

### PR DESCRIPTION
**Pull request by mymattcarroll**
### Implementation

- [ ] Have you tested your implementation locally?
- [ ] If applicable, has an appropriate changelog entry been added? Do not include if you are fixing/changing something that is only in the current release.
- [ ] There are no warnings that have been suppressed unnecessarily
- [ ] Have automated tests been added, or have related ones been updated to cover the change?
- [ ] Have all OneBlink dependency updates been completed (eg apps / apps-react / types etc).
- [ ] Have you ensured this change does not add unwanted dependencies?
- [ ] If this PR contains a refactor, have relevant Jira testing tasks added?
- [ ] Changes that will knowingly make the feature/bug incomplete have been commented with TODO and a description of what needs to be done to finish the feature/bug
- [ ] Any changes made to public APIs have been reflected in the documentation
- [ ] Have you isolated business logic where possible to allow for unit testing?

### Logging and Debugging

- [ ] Are the error messages, if any, informative?
- [ ] Are there enough log events and are they written in a way that allows for easy debugging?
- [ ] "Debugging" code removed
- [ ] Front-end: No erroneous `Console.WriteLines`

### Readability

- [ ] All class, variable, property and method modifiers are provided with the smallest scope possible
- [ ] New files, variables and functions are descriptive/comprehensible and named consistently.
- [ ] There is no dead code (unreachable code)
- [ ] There is no usage of [magic numbers](<https://en.wikipedia.org/wiki/Magic_number_(programming)>)
- [ ] There is no commented out code.
- [ ] In hard-to-understand areas, comments exist and describe rationale or reasons for decisions in code

### Security

- [ ] All personal data inputs are checked (for the correct type, length/size, format, and range).
- [ ] No sensitive information is logged or visible in a stacktrace
- [ ] Are authorization and authentication handled correctly?
- [ ] Is (user) input validated, sanitized, and escaped to prevent security attacks such as cross-site scripting or SQL injection?
- [ ] Is data retrieved from external APIs or libraries checked for security issues?
- [ ] Do API endpoints return appropriate status codes

## Reviewer Guide

- It is important that you understand the purpose of the PR.
- You are encouraged to engage with the requestor if you do not understand any of the proposed code changes/additions/deletions.
- Do you, the reviewer, understand what the code does? Do you think a specific expert, like a security expert or a usability expert, should look over the code before it can be accepted?
- Is a framework, API, library, or service used that should not be used? Are there alternatives you could recommend?
- Are there existing hooks/components/functions in the same code base that could be utilised?
- When reviewing tests, attempt to identify missing edge cases that may be relevant to the proposed implementation
- Ensure you check for:
  - Security
  - Scalability
  - Performance
  - Maintainability

### CODE REVIEW

**Review by divporter — APPROVED** [review]

**Review by jdawg093 — APPROVED** [review]

**Review by kizaonline — APPROVED** [review]

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 ## Requester Checklist
 
+Please only check the items that you have done. Do not check items that are not applicable to your PR.
```
Comment (RyanButton) [inline .github/pull_request_template.md]: How far do we want to take this? E.g `Have all OneBlink dependency updates been completed (eg apps / apps-react / types etc)` => if 0 dependencies need updating, then doing 0 of them is a tick?

Comment (mymattcarroll) [reply .github/pull_request_template.md]: 0 dependencies === not applicable IMO

Anyone else have an opinion?

Comment (mymattcarroll) [reply .github/pull_request_template.md]: Another example:

if your PR contains no numbers (hard coded or in variables), their is no need to tick:

> There is no usage of [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming))

Comment (RyanButton) [reply .github/pull_request_template.md]: Examples sound good. Potentially could help highlight the subtle difference with something this as well: 
```suggestion
Please only check the items that you have actioned. Do not check items that are not applicable to your PR.
```

Comment (Zaxist) [reply .github/pull_request_template.md]: The ideal way to handle some of these scenarios (in my opinion), would be to have checkboxes and sub checkboxes. Eg.

- Does this PR require OB dependency updates (eg apps / apps-react / types etc)?
  - Have you updated all OB dependencies?

- Does this PR include changes that will knowingly make the feature/bug incomplete?
  - Have they been commented with TODO and a description of what needs to be done to finish the feature/bug?

Otherwise you will always have situations where it isn't clear for people whether something N/A should be checked or not.

Comment (Zaxist) [reply .github/pull_request_template.md]: If that's possible in MD, I assume it probably is.

Comment (Zaxist) [reply .github/pull_request_template.md]: Also, I don't expect you to do this right now, but it does seem like it should be the ultimate goal to me.

**Review by Zaxist — APPROVED** [review]
